p32/src/shapes.rs

```diff
@@ -176,17 +176,40 @@ impl Shape for DynamicShape {
     }
 }
 
-// pub enum Return {
-//     First,
-//     Second,
-// }
-
-// pub fn bigger_area_to_perimeter<'a>(first: &'a [impl Shape], second: &'a [impl Shape]) -> Return {
-//     if first.area_to_perimeter() > second.area_to_perimeter() {
-//         println!("{first:#?}");
-//         Return::First(first[0])
-//     } else {
-//         println!("{second:#?}");
-//         Return::Second
-//     }
-// }
+pub enum Return<'a, T, U> {
+    First(&'a T),
+    Second(&'a U),
+}
+
+pub fn bigger_area_to_perimeter<'a>(
+    first: &'a [impl Shape],
+    second: &'a [impl Shape],
+) -> Return<'a, impl Shape, impl Shape> {
```

Comment:
In this case it's better to use `fn biggest_area_to_perimeter<'a, 'b, T: Shape, U: Shape>(first: &[T], second: &[U]) -> Return<'a, 'b, T, U>`.